README.md

```diff
@@ -0,0 +1,1116 @@
+sed -i '' 's/unconditional_peer_ids = ""/unconditional_peer_ids = "'$VAL_2_NODE_ID'"/g' $VAL_1_CHAIN_DIR/config/config.toml
+sed -i '' 's/pprof_laddr = "localhost:6060"/pprof_laddr = "localhost:'$VAL_2_PPROF_PORT'"/g' $VAL_2_CHAIN_DIR/config/config.toml
+sed -i '' 's/addr_book_strict = true/addr_book_strict = false/g' $VAL_2_CHAIN_DIR/config/config.toml
+sed -i '' 's/addr_book_strict = true/addr_book_strict = false/g' $VAL_1_CHAIN_DIR/config/config.toml
+```
+disable the Rosetta API server
+```shell
+# Enable defines if the Rosetta API server should be enabled.
+# enable = false
+sed -i '' '/Enable defines if the Rosetta API server/,/Address defines the Rosetta API server/s/enable = true/enable = false/' $VAL_1_CHAIN_DIR/config/app.toml
+sed -i '' '/Enable defines if the Rosetta API server/,/Address defines the Rosetta API server/s/enable = true/enable = false/' $VAL_2_CHAIN_DIR/config/app.toml
+```
+### Cosmosvisor
+#### set cosmosvisor
+create the folder for cosmosvisor for val1 and val2, and put the old binary in `cosmovisor/genesis/bin`.
+```shell
+mkdir -p $VAL_1_CHAIN_DIR/cosmovisor/genesis/bin
+mkdir -p $VAL_2_CHAIN_DIR/cosmovisor/genesis/bin
+cp $(which gaiad) $VAL_1_CHAIN_DIR/cosmovisor/genesis/bin
+cp $(which gaiad) $VAL_2_CHAIN_DIR/cosmovisor/genesis/bin
+```
+
+Build the new gaia binary
+```shell
+cd gaia
+git checkout start-upgrade
+make install
+```
+Create the folders for the two nodes and put the upgrade gaia binary into `cosmovisor/upgrades/Vega/bin`:
+```shell
+mkdir -p $VAL_1_CHAIN_DIR/cosmovisor/upgrades/Vega/bin
+mkdir -p $VAL_2_CHAIN_DIR/cosmovisor/upgrades/Vega/bin
+cp $(which gaiad) $VAL_1_CHAIN_DIR/cosmovisor/upgrades/Vega/bin
+cp $(which gaiad) $VAL_2_CHAIN_DIR/cosmovisor/upgrades/Vega/bin
+```
+#### Start by cosmovisor
+For val1:
+```shell
+export DAEMON_NAME=gaiad
+export DAEMON_HOME= $(pwd)/$VAL_1_CHAIN_DIR
+export DAEMON_RESTART_AFTER_UPGRADE=true
+cosmovisor start --x-crisis-skip-assert-invariants --home $VAL_1_CHAIN_DIR
+```
+For val2:
+
+open a new terminal:
+```shell
+export DAEMON_NAME=gaiad
+export DAEMON_HOME= $(pwd)/$VAL_2_CHAIN_DIR
+export DAEMON_RESTART_AFTER_UPGRADE=true
+cosmovisor start --x-crisis-skip-assert-invariants --home $VAL_2_CHAIN_DIR
+```
+### Propose upgrade
+The user owns by val2 is a delegator. So user can vote. Since we changed the [gov parameters](####Modify some gov parameters for test efficiency), the delegations this user delegated are far enough for this proposal to pass. 
+```shell
+cosmovisor tx gov submit-proposal software-upgrade Vega \
+--title Vega \
+--deposit 100uatom \
+--upgrade-height 7368587 \
+--upgrade-info "upgrade to Vega" \
+--description "upgrade to Vega" \
+--gas 400000 \
+--from user \
+--keyring-backend test \
+--chain-id test \
+--home data/test/val2 \
+--node tcp://localhost:36657 \
+--yes
+```
+### Vote
+open a new terminal to vote by user.
+```shell
+cd vega-test
+gaiad query gov proposal 54 \
```

Comment:
this should be:

`gaiad tx gov vote 54 yes --from user --keyring-backend test --chain-id test --home data/test/val2 --node tcp://127.0.0.1:36657`